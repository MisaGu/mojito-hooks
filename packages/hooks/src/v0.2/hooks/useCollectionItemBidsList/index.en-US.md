---
nav:
  path: /hooks
  group: Mojito
---

# useCollectionItemBidsList

Returns the selected collection item, including its bids (but not `remainingCount`).

Useful when working with auction lots.

## Examples

### Default usage

<code src="./demo/demo1.tsx" />
