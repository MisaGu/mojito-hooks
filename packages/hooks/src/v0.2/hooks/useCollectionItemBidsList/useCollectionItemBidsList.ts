import { EMojitoQueries, IUseQueryResult } from '../../domain/gql/queries';
import { IMojitoCollectionItemDetailsBid } from '../../domain/interfaces';
import { useCollection } from '../useCollection/useCollection';
import { useMojitoFactory } from '../useMojitoFactory/useMojitoFactory';

export function useCollectionItemBidsList(
  id: string,
  _slug?: string,
): {
  bids: IMojitoCollectionItemDetailsBid[];
  bidsLoading: boolean;
  bidsError: IUseQueryResult['error'];
  bidsRefetch: () => void;
} {
  const { collection } = useCollection();
  const slug = collection?.slug;
  const { data, error, loading, refetch } = useMojitoFactory({
    query: EMojitoQueries.collectionItemByIdBidsList,
    variables: { id, slug: _slug ?? slug },
  });

  // if (error) console.error(error);

  return {
    bids: data?.details?.bids,
    bidsLoading: loading,
    bidsError: error,
    bidsRefetch: refetch,
  };
}
