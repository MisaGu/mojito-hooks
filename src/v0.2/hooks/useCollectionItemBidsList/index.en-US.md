---
nav:
  path: /hooks
  group: Mojito
---

# useCollectionItemBidsList

Returns the selected collection item, including its bids (but not `remainingCount`).

Useful when working with auction lots.

## Examples

<code src="./demo/demo1.tsx" />
