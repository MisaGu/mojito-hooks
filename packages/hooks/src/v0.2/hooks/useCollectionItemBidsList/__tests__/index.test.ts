import { useCollectionItemBidsList } from '../useCollectionItemBidsList';

describe('useCollectionItemBidsList', () => {
  it('should be defined', () => {
    expect(useCollectionItemBidsList).toBeDefined();
  });
});
