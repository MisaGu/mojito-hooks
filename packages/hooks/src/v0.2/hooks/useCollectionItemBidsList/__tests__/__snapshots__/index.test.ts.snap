// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useCollectionItemBidsList() returns the right result when needed 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "itemBids": Object {
    "details": Object {
      "bids": Array [],
      "endDate": "",
      "id": "mock-item-id",
      "startDate": "",
    },
    "id": "mock-item-id",
  },
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": false,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
}
`;
